### PR TITLE
Reboot test with USB

### PR DESCRIPTION
**Pull request by dautovri**
Assumes a USB controller. Key purpose is to verify that a reboot (sudo shutdown -r +1) from the guest doesn’t disrupt the set of assigned adapters.

### CODE REVIEW

File: tests/hardware_reboot/testdata/test_run.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+exec -t 20m bash ssh.sh 8027
+stdout 'Ubuntu'
+
+exec -t 20m bash get-usb.sh 8027
+grep 'QEMU USB HARDDRIVE' {{$usb_dev}}.usb.product
+
+# restart pod
+eden pod stop n1
+exec sleep 20
+eden pod start n1
```
Comment (giggsoff) [inline tests/hardware_reboot/testdata/test_run.txt:21]: How does this compare with the description of the PR? I mean "reboot **from the guest**".

**Review by giggsoff — CHANGES_REQUESTED** [review]
Please squash and rebase your commits and take a look onto my comments.

**Comment by sadov** [comment]
LGTM

**Comment by sadov** [comment]
@dautovri seems wise to rename `hardware_reboot` to something USB related -- we'll have a lot of reboot tests. And `usb-pt_test` is no longer needed -- `hardware_reboot` repeating its functionality. Just replace it and remove that tricky numbering -- 22.1/2.

**Comment by sadov** [comment]
LGTM `eden test tests/workflow/ -v debug` passed

**Comment by mydatascience** [comment]
@dautovri  Do we have an explicit error if the USB controller is not present. 
Say the message "USB not present"

File: tests/hardware_reboot/testdata/hardware_usb_reboot.txt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+exec -t 20m bash get-usb.sh 2223
+grep 'QEMU USB HARDDRIVE' {{$usb_dev}}.usb.product
+
+# reboot from guest
+exec -t 20m bash reboot.sh 2223
+exec sleep 10
+
+exec -t 20m bash ssh.sh 2223
+stdout 'Ubuntu'
```
Comment (eriknordmark) [inline tests/hardware_reboot/testdata/hardware_usb_reboot.txt:26]: How do you know it rebooted? 
FWIW in the ztests we check last (last -n 1 --time-format iso reboot | awk '{print $5}'| head -1) before the reboot and then wait for that timestamp to increase.

File: tests/hardware_reboot/testdata/hardware_usb_reboot.txt [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Simple test of USB passthrough functionality after reboot of guest
+
+{{$usb_dev := "2-2"}}
```
Comment (eriknordmark) [inline tests/hardware_reboot/testdata/hardware_usb_reboot.txt:3]: Does this work with Xen?

File: tests/hardware_reboot/testdata/hardware_usb_reboot.txt (hunk 2 above)
Comment (zed-sadov) [reply tests/hardware_reboot/testdata/hardware_usb_reboot.txt:26]: Actually we have a special reboot-detector eden.reboot.test (tests/reboot/reboot_test.go), which we using for ex. in tests/workflow/testdata/reboot_test.txt. In this detector we checking the lastRebootTime from dinfo. The main advantage is that we don't need SSH access.

File: tests/hardware_reboot/testdata/hardware_usb_reboot.txt (hunk 3 above)
Comment (giggsoff) [reply tests/hardware_reboot/testdata/hardware_usb_reboot.txt:3]: Good point. I see it on another address inside.
```
root@0ce327c6-7f9e-433d-b73f-9b7e5a096cdd:~# cat /sys/bus/usb/devices/1-1/product 
QEMU USB HARDDRIVE
```

File: tests/hardware_reboot/testdata/hardware_usb_reboot.txt (hunk 2 above)
Comment (dautovri) [reply tests/hardware_reboot/testdata/hardware_usb_reboot.txt:26]: @sadov @eriknordmark Good point, I will apply checks.

**Comment by dautovri** [comment]
https://github.com/lf-edge/eden/pull/560 changes

**Pull request merged**